lib/ChunkGraph.js

```diff
@@ -568,11 +574,24 @@ class ChunkGraph {
 	getChunkModulesIterableBySourceType(chunk, sourceType) {
 		const cgc = this._getChunkGraphChunk(chunk);
 		const modulesWithSourceType = cgc.modules
-			.getFromUnorderedCache(modulesBySourceType)
+			.getFromUnorderedCache(modulesBySourceType(cgc.sourceTypesByModule))
```

Comment:
This breaks the caching. You need to pass the same identify.

Reply:
e. g. you could store `this.modulesBySourceType = modulesBySourceType(undefined)` in the `ChunkGraphChunk` constructor.